### PR TITLE
Remove QT

### PR DESCRIPTION
**Pull request by Dexterp37**
This removes QT and rolls back some of the changes that were made to the APIs in order to support the QT platform (FYI @brizental as your team was the only consumer of this API).

TODO:

- [x] Add a breaking change to the changelog
- [x] Verify that everything else works in a product
- [x] Check if it's ok to remove the unsafe `generateUUIDv4` fallback that was introduced due to Qt. **(It was not ok, as we might still need this for webextensions that this version still supports)**.

### Pull Request checklist ###

- [ ] **Quality**: Make sure this PR builds and runs cleanly.
  - Inside the `glean/` folder, run:
    - `npm run test` Runs _all_ tests
    - `npm run lint` Runs _all_ linters
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry to `CHANGELOG.md` or an explanation of why it does not need one
- [ ] **Documentation**: This PR includes documentation changes, an explanation of why it does not need that or a follow-up bug has been filed to do that work
  - Documentation should be added to [The Glean Book](https://mozilla.github.io/glean/book/index.html) when making changes to Glean's user facing API
    - When changes to the Glean Book are necessary, link to the corresponding PR on the [`mozilla/glean`](https://github.com/mozilla/glean) repository
  - Documentation should be added to [the Glean.js developer documentation](https://github.com/mozilla/glean.js/tree/main/docs) when making internal code changes

### CODE REVIEW

File: automation/size/docs.js [hunk 1]
```diff
@@ -1,127 +1,125 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (travis79) [inline automation/size/docs.js:1]: It's weird that it thinks you changed this entire file when it's mostly the same. Did the line endings change or something?

**Review by travis79 — COMMENTED** [review]
This is awesome! I'm a little concerned about the files that appear to have every line changed, did we change line endings from Linux-y to Windows-y maybe?

**Review by travis79 — APPROVED** [review]

**Review by rosahbruno — APPROVED** [review]
It looks like we may be able to drop the `sqlite` dev dependency in the `/glean` `package.json` since it was only being used by QT.

Also looks like there may be some KnownOperatingSystems we may be able to drop now, but probably not a big deal if we didn't.
https://github.com/mozilla/glean.js/blob/main/glean/src/core/platform_info/shared.ts#L23-L26

Besides that, everything looks good. I will try and run this in a project and make sure everything is working as expected.

**Comment by rosahbruno** [comment]
@Dexterp37 
I ran this version of Glean with the debug ping viewer via this branch https://github.com/rosahbruno/glean.js/tree/remove-qt-dist and everything worked as expected 👍

**Comment by Dexterp37** [comment]
> It looks like we may be able to drop the `sqlite` dev dependency in the `/glean` `package.json` since it was only being used by QT.

Good call, remove that!

> Also looks like there may be some KnownOperatingSystems we may be able to drop now, but probably not a big deal if we didn't. https://github.com/mozilla/glean.js/blob/main/glean/src/core/platform_info/shared.ts#L23-L26

Yes, saw that and thought to remove it as I wasn't really sure and the risk didn't seem worth it.

> Besides that, everything looks good. I will try and run this in a project and make sure everything is working as expected.

Thanks!

**Pull request merged**